chore(Node): Do not keep the entire genesis file in memory during node operation (#4247)
cometbft/cometbft: .changelog/unreleased/breaking-changes/4250-node-no-genesis.md

> - `[Node]` no longer stores a pointer to a `types.GenesisDoc` after completing its

Reviewer: ```suggestion
- `[node]` Don't store a pointer to a `types.GenesisDoc` after the node is running
```
